### PR TITLE
MM-9556 Added documentation for new file upload route

### PR DESCRIPTION
**Pull request by hmhealey**
This is for the new file upload API required for the iOS share extension. Since it's documentation, it's pretty self-explanatory

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-9556

### CODE REVIEW

File: v4/source/files.yaml [hunk 1]
```diff
@@ -5,29 +5,45 @@
       summary: Upload a file
       description: |
         Uploads a file that can later be attached to a post.
+
+        This request can either be a multipart/form-data request with a channel_id, files and optional
+        client_ids defined in the FormData, or it can be a request with the channel_id and filename
+        defined as query parameters with the contents of a single file in the body of the request.
```
Comment (enahum) [inline v4/source/files.yaml]: isn't this alternative request type suppose to be of content type `application/octet-stream` ?

Comment (hmhealey) [reply v4/source/files.yaml]: It can be any type. The server doesn't really care what it is as long as it's not multipart

Comment (jwilander) [reply v4/source/files.yaml]: I'm 0/5 here, it's pretty common not to send a content-type when you're just sending bytes

File: v4/source/files.yaml [hunk 2]
```diff
@@ -5,29 +5,45 @@
       summary: Upload a file
       description: |
         Uploads a file that can later be attached to a post.
+
+        This request can either be a multipart/form-data request with a channel_id, files and optional
```
Comment (jwilander) [inline v4/source/files.yaml]: Actually, can you add a note here about the new functionality only being available in 4.8+ servers?

**Pull request merged**